<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Free Space: Visualisation of scRNA sequencing using Python | Portfolio</title>
  <meta name="description" content="3 Free Space: Visualisation of scRNA sequencing using Python | Portfolio" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Free Space: Visualisation of scRNA sequencing using Python | Portfolio" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Free Space: Visualisation of scRNA sequencing using Python | Portfolio" />
  
  
  

<meta name="author" content="AK" />


<meta name="date" content="2025-01-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cv.html"/>
<link rel="next" href="guerrilla-analytics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="cv.html"><a href="cv.html"><i class="fa fa-check"></i><b>2</b> CV</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cv.html"><a href="cv.html#rd-year"><i class="fa fa-check"></i><b>2.1</b> 3rd year</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="cv.html"><a href="cv.html#theoretical-courses"><i class="fa fa-check"></i><b>2.1.1</b> Theoretical courses</a></li>
<li class="chapter" data-level="2.1.2" data-path="cv.html"><a href="cv.html#practical-courses"><i class="fa fa-check"></i><b>2.1.2</b> Practical courses</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="cv.html"><a href="cv.html#honours-projects-extracurricular"><i class="fa fa-check"></i><b>2.2</b> Honours projects (extracurricular)</a></li>
<li class="chapter" data-level="2.3" data-path="cv.html"><a href="cv.html#nd-year"><i class="fa fa-check"></i><b>2.3</b> 2nd year</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="cv.html"><a href="cv.html#theoretical-courses-1"><i class="fa fa-check"></i><b>2.3.1</b> Theoretical courses</a></li>
<li class="chapter" data-level="2.3.2" data-path="cv.html"><a href="cv.html#practical-courses-1"><i class="fa fa-check"></i><b>2.3.2</b> Practical courses</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="cv.html"><a href="cv.html#extracurricular-courses"><i class="fa fa-check"></i><b>2.4</b> Extracurricular courses</a></li>
<li class="chapter" data-level="2.5" data-path="cv.html"><a href="cv.html#work-experience"><i class="fa fa-check"></i><b>2.5</b> Work Experience</a></li>
<li class="chapter" data-level="2.6" data-path="cv.html"><a href="cv.html#internship"><i class="fa fa-check"></i><b>2.6</b> Internship</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="free-space-visualisation-of-scrna-sequencing-using-python.html"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html"><i class="fa fa-check"></i><b>3</b> Free Space: Visualisation of scRNA sequencing using Python</a>
<ul>
<li class="chapter" data-level="3.1" data-path="free-space-visualisation-of-scrna-sequencing-using-python.html"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#scrna-sequencing-using-python"><i class="fa fa-check"></i><b>3.1</b> scRNA sequencing using Python</a></li>
<li class="chapter" data-level="3.2" data-path="free-space-visualisation-of-scrna-sequencing-using-python.html"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#planning"><i class="fa fa-check"></i><b>3.2</b> Planning</a></li>
<li class="chapter" data-level="3.3" data-path="free-space-visualisation-of-scrna-sequencing-using-python.html"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#freespace-uitwerking"><i class="fa fa-check"></i><b>3.3</b> Freespace: uitwerking</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="guerrilla-analytics.html"><a href="guerrilla-analytics.html"><i class="fa fa-check"></i><b>4</b> Guerrilla analytics</a></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible Research</a>
<ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research.html"><a href="reproducible-research.html#c.elegans-plate-experiment"><i class="fa fa-check"></i><b>5.1</b> <em>C.elegans</em> plate experiment</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="open-peer-review.html"><a href="open-peer-review.html"><i class="fa fa-check"></i><b>6</b> Open peer review</a>
<ul>
<li class="chapter" data-level="6.1" data-path="open-peer-review.html"><a href="open-peer-review.html#part-1"><i class="fa fa-check"></i><b>6.1</b> Part 1</a></li>
<li class="chapter" data-level="6.2" data-path="open-peer-review.html"><a href="open-peer-review.html#part-2"><i class="fa fa-check"></i><b>6.2</b> Part 2</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="making-an-r-package.html"><a href="making-an-r-package.html"><i class="fa fa-check"></i><b>7</b> Making an R Package</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="free-space-visualisation-of-scrna-sequencing-using-python" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Free Space: Visualisation of scRNA sequencing using Python<a href="free-space-visualisation-of-scrna-sequencing-using-python.html#free-space-visualisation-of-scrna-sequencing-using-python" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The goal of the free space is to develop skills that are not discussed in this course. Dit biedt veel vrijheid en daarnaast ook de ruimte om jezelf te ontwikkelen in de richting van een latere baan.</p>
<div id="scrna-sequencing-using-python" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> scRNA sequencing using Python<a href="free-space-visualisation-of-scrna-sequencing-using-python.html#scrna-sequencing-using-python" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In februari zal ik stage lopen bij de DiMicheleLAB, in cambridge. In onze gesprekken hebben zij laten weten dat ik python en R zal gebruiken bij het verwerken van de data die ik in mijn stage zal krijgen. Ik wil daarom deze vrije ruimte gebruiken om mij alvast te verdiepen in het visualiseren van data in R. Al snel komt singlecell RNA sequencing naar boven wanneer ik zoek op verschillende analyses. Echter is het niet mogelijk om in de korte tijd, 32 uur, de hele workflow van de scRNA sequencing te leren. Ik ga me daarom focussen op de visualisatie van de data met behulp van R, specifiek de packages <strong>tidyverse</strong>, <strong>Seurat</strong>, <strong>ComplexHeatmap</strong> en <strong>circlize</strong>.</p>
</div>
<div id="planning" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Planning<a href="free-space-visualisation-of-scrna-sequencing-using-python.html#planning" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Het is de bedoeling dat ik elke week 1 dag, zo’n 7 uur, besteed aan het vrije deel. Dit bekent dat er ~5 dagen besteed worden of dat de “tijd” bij elkaar optelt tot 32 uur.</p>
<div id="xqsmrjpkiy" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#xqsmrjpkiy table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#xqsmrjpkiy thead, #xqsmrjpkiy tbody, #xqsmrjpkiy tfoot, #xqsmrjpkiy tr, #xqsmrjpkiy td, #xqsmrjpkiy th {
  border-style: none;
}

#xqsmrjpkiy p {
  margin: 0;
  padding: 0;
}

#xqsmrjpkiy .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#xqsmrjpkiy .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#xqsmrjpkiy .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#xqsmrjpkiy .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#xqsmrjpkiy .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xqsmrjpkiy .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xqsmrjpkiy .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xqsmrjpkiy .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#xqsmrjpkiy .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#xqsmrjpkiy .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#xqsmrjpkiy .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#xqsmrjpkiy .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#xqsmrjpkiy .gt_spanner_row {
  border-bottom-style: hidden;
}

#xqsmrjpkiy .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#xqsmrjpkiy .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#xqsmrjpkiy .gt_from_md > :first-child {
  margin-top: 0;
}

#xqsmrjpkiy .gt_from_md > :last-child {
  margin-bottom: 0;
}

#xqsmrjpkiy .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#xqsmrjpkiy .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#xqsmrjpkiy .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#xqsmrjpkiy .gt_row_group_first td {
  border-top-width: 2px;
}

#xqsmrjpkiy .gt_row_group_first th {
  border-top-width: 2px;
}

#xqsmrjpkiy .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xqsmrjpkiy .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#xqsmrjpkiy .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#xqsmrjpkiy .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xqsmrjpkiy .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xqsmrjpkiy .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#xqsmrjpkiy .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#xqsmrjpkiy .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#xqsmrjpkiy .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xqsmrjpkiy .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xqsmrjpkiy .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#xqsmrjpkiy .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xqsmrjpkiy .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#xqsmrjpkiy .gt_left {
  text-align: left;
}

#xqsmrjpkiy .gt_center {
  text-align: center;
}

#xqsmrjpkiy .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#xqsmrjpkiy .gt_font_normal {
  font-weight: normal;
}

#xqsmrjpkiy .gt_font_bold {
  font-weight: bold;
}

#xqsmrjpkiy .gt_font_italic {
  font-style: italic;
}

#xqsmrjpkiy .gt_super {
  font-size: 65%;
}

#xqsmrjpkiy .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#xqsmrjpkiy .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#xqsmrjpkiy .gt_indent_1 {
  text-indent: 5px;
}

#xqsmrjpkiy .gt_indent_2 {
  text-indent: 10px;
}

#xqsmrjpkiy .gt_indent_3 {
  text-indent: 15px;
}

#xqsmrjpkiy .gt_indent_4 {
  text-indent: 20px;
}

#xqsmrjpkiy .gt_indent_5 {
  text-indent: 25px;
}

#xqsmrjpkiy .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#xqsmrjpkiy div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Datum">Datum</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Tijd">Tijd</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Inhoud">Inhoud</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="Datum" class="gt_row gt_left">12 november</td>
<td headers="Tijd" class="gt_row gt_left">2 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Vinden van een onderwerp</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">19 november</td>
<td headers="Tijd" class="gt_row gt_left">1 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Python tutorial</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">21 november</td>
<td headers="Tijd" class="gt_row gt_left">4.5 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Opzetten van de juiste omgeving</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">21 november</td>
<td headers="Tijd" class="gt_row gt_left">2 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Tutorial; Start met hoofdstuk 1: Doublet removal</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">21 november</td>
<td headers="Tijd" class="gt_row gt_left">6 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Tutorial; Hoofdstuk 2: Preprocessing</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">25 november</td>
<td headers="Tijd" class="gt_row gt_left">1 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Vinden van een nieuw onderwerp</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">25 november</td>
<td headers="Tijd" class="gt_row gt_left">2 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Opzetten van de juiste omgeving</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">30 november</td>
<td headers="Tijd" class="gt_row gt_left">3 uur</td>
<td headers="Inhoud" class="gt_row gt_left">tutorial: Single-cell RNAseq data visualization in R</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">30 november</td>
<td headers="Tijd" class="gt_row gt_left">2 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Opzoeken van een nieuwe dataset</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">2 december</td>
<td headers="Tijd" class="gt_row gt_left">3 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Toepassen van de tutorial op nieuwe data</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">4 december</td>
<td headers="Tijd" class="gt_row gt_left">3 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Toepassen van de tutorial op nieuwe data</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">10 december</td>
<td headers="Tijd" class="gt_row gt_left">4 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Toepassen van de tutorial op nieuwe data</td></tr>
    <tr><td headers="Datum" class="gt_row gt_left">11 december</td>
<td headers="Tijd" class="gt_row gt_left">2 uur</td>
<td headers="Inhoud" class="gt_row gt_left">Toepassen van de tutorial op nieuwe data</td></tr>
  </tbody>
  
  
</table>
</div>

</div>
<div id="freespace-uitwerking" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Freespace: uitwerking<a href="free-space-visualisation-of-scrna-sequencing-using-python.html#freespace-uitwerking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Replicating the graphs as shown in the paper “TREM2 protects from atherosclerosis by limiting necrotic core formation”, Piollet et al. (2024). doi: 10.1038/s44161-024-00429-9.</p>
<p>Atherosclerose, veroorzaakt door de ophoping van lipiden en ontstekingen in de vaatwanden, leidt tot complicaties zoals hartaanvallen en beroertes. De lipide-sensor receptor TREM2, die sterk tot uiting komt in macrofaagschuimcellen, speelt een cruciale rol in atherosclerose. Deze studie toont aan dat TREM2-deficiëntie de vorming van necrotische kernen vergroot, terwijl activering van TREM2 deze juist vermindert in vroege atherosclerose. TREM2 is essentieel voor de efferocytose en het overleven van met lipiden beladen macrofagen, wat de balans bewaart tussen schuimcelsterfte en opruiming van dode cellen in plaques. Dit onderstreept de belangrijke rol van TREM2 in het verminderen van plaque-necrose en het behouden van vaatgezondheid bij atherosclerose.</p>
<p>Single-cell RNA-sequencing and analysis was done on aortic cells in Ldlr−/− mice.</p>
<p>Sidenote: de messages/warnings die geproduceerd worden wanneer DoubletFinder gerund wordt, kunnen niet onderdrukt worden. Dit resulteerde in hele lange, onnodige chucks in de output. Dit kan negeert worden.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-1" tabindex="-1"></a>packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Seurat&quot;</span>, <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;viridis&quot;</span>, <span class="st">&quot;RColorBrewer&quot;</span>, <span class="st">&quot;plyr&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, </span>
<span id="cb1-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-2" tabindex="-1"></a>              <span class="st">&quot;cowplot&quot;</span>, <span class="st">&quot;tidyr&quot;</span>, <span class="st">&quot;tidyverse&quot;</span>, <span class="st">&quot;dittoSeq&quot;</span>, <span class="st">&quot;DropletUtils&quot;</span>, </span>
<span id="cb1-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-3" tabindex="-1"></a>              <span class="st">&quot;DoubletFinder&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;patchwork&quot;</span>, <span class="st">&quot;pacman&quot;</span>)</span>
<span id="cb1-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-5" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> packages) {</span>
<span id="cb1-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-6" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(pkg, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-7" tabindex="-1"></a>    <span class="fu">install.packages</span>(pkg)</span>
<span id="cb1-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-8" tabindex="-1"></a>  }</span>
<span id="cb1-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-9" tabindex="-1"></a>}</span>
<span id="cb1-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-11" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(Seurat, ggplot2, viridis, RColorBrewer, plyr, dplyr, </span>
<span id="cb1-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-12" tabindex="-1"></a>               cowplot, tidyr, tidyverse, dittoSeq, DropletUtils, </span>
<span id="cb1-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb1-13" tabindex="-1"></a>               DoubletFinder, knitr, patchwork)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-1" tabindex="-1"></a><span class="co">#Inladen van data </span></span>
<span id="cb2-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-2" tabindex="-1"></a>B11.data <span class="ot">=</span> <span class="fu">Read10X</span>(<span class="st">&quot;/Users/annemariekekreeft/school/sc_rna_seq_oefen/TREM2_atherosclerose_Piollet/raw_data/B11/outs/filtered_feature_bc_matrix&quot;</span>)</span>
<span id="cb2-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-3" tabindex="-1"></a>B12.data <span class="ot">=</span> <span class="fu">Read10X</span>(<span class="st">&quot;/Users/annemariekekreeft/school/sc_rna_seq_oefen/TREM2_atherosclerose_Piollet/raw_data/B12/outs/filtered_feature_bc_matrix&quot;</span>)</span>
<span id="cb2-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-4" tabindex="-1"></a>C1.data <span class="ot">=</span> <span class="fu">Read10X</span>(<span class="st">&quot;/Users/annemariekekreeft/school/sc_rna_seq_oefen/TREM2_atherosclerose_Piollet/raw_data/C1/outs/filtered_feature_bc_matrix&quot;</span>)</span>
<span id="cb2-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-5" tabindex="-1"></a>C2.data <span class="ot">=</span> <span class="fu">Read10X</span>(<span class="st">&quot;/Users/annemariekekreeft/school/sc_rna_seq_oefen/TREM2_atherosclerose_Piollet/raw_data/C2/outs/filtered_feature_bc_matrix&quot;</span>)</span>
<span id="cb2-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-6" tabindex="-1"></a>C3.data <span class="ot">=</span> <span class="fu">Read10X</span>(<span class="st">&quot;/Users/annemariekekreeft/school/sc_rna_seq_oefen/TREM2_atherosclerose_Piollet/raw_data/C3/outs/filtered_feature_bc_matrix&quot;</span>)</span>
<span id="cb2-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-7" tabindex="-1"></a>C4.data <span class="ot">=</span> <span class="fu">Read10X</span>(<span class="st">&quot;/Users/annemariekekreeft/school/sc_rna_seq_oefen/TREM2_atherosclerose_Piollet/raw_data/C4/outs/filtered_feature_bc_matrix&quot;</span>)</span>
<span id="cb2-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-9" tabindex="-1"></a><span class="co"># Maken van Seurat objecten </span></span>
<span id="cb2-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-10" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span>B11.data)</span>
<span id="cb2-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-11" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span>B12.data)</span>
<span id="cb2-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-12" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span>C1.data)</span>
<span id="cb2-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-13" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span>C2.data)</span>
<span id="cb2-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-14" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span>C3.data)</span>
<span id="cb2-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-15" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span>C4.data)</span>
<span id="cb2-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-16" tabindex="-1"></a></span>
<span id="cb2-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-17" tabindex="-1"></a><span class="co"># Normalize RNA data with log normalization</span></span>
<span id="cb2-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-18" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">NormalizeData</span>(B11)</span>
<span id="cb2-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-19" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">NormalizeData</span>(B12)</span>
<span id="cb2-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-20" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">NormalizeData</span>(C1)</span>
<span id="cb2-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-21" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">NormalizeData</span>(C2)</span>
<span id="cb2-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-22" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">NormalizeData</span>(C3)</span>
<span id="cb2-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-23" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">NormalizeData</span>(C4)</span>
<span id="cb2-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-24" tabindex="-1"></a></span>
<span id="cb2-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-25" tabindex="-1"></a><span class="do">## **Quality Control Filtering**</span></span>
<span id="cb2-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-26" tabindex="-1"></a><span class="co">#Add the &quot;percent.mt&quot; data</span></span>
<span id="cb2-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-27" tabindex="-1"></a>B11[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">=</span> <span class="fu">PercentageFeatureSet</span>(B11, <span class="at">pattern =</span> <span class="st">&quot;^mt-&quot;</span>)</span>
<span id="cb2-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-28" tabindex="-1"></a>B12[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">=</span> <span class="fu">PercentageFeatureSet</span>(B12, <span class="at">pattern =</span> <span class="st">&quot;^mt-&quot;</span>)</span>
<span id="cb2-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-29" tabindex="-1"></a>C1[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">=</span> <span class="fu">PercentageFeatureSet</span>(C1, <span class="at">pattern =</span> <span class="st">&quot;^mt-&quot;</span>)</span>
<span id="cb2-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-30" tabindex="-1"></a>C2[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">=</span> <span class="fu">PercentageFeatureSet</span>(C2, <span class="at">pattern =</span> <span class="st">&quot;^mt-&quot;</span>)</span>
<span id="cb2-31"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-31" tabindex="-1"></a>C3[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">=</span> <span class="fu">PercentageFeatureSet</span>(C3, <span class="at">pattern =</span> <span class="st">&quot;^mt-&quot;</span>)</span>
<span id="cb2-32"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb2-32" tabindex="-1"></a>C4[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">=</span> <span class="fu">PercentageFeatureSet</span>(C4, <span class="at">pattern =</span> <span class="st">&quot;^mt-&quot;</span>)</span></code></pre></div>
<p>Firstly, the data was normalised and plotted. To increase quality of the data, a subset was taken (&lt;10% mt).</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-1" tabindex="-1"></a><span class="co"># First subset &lt;10% mt</span></span>
<span id="cb3-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-2" tabindex="-1"></a><span class="co"># Filtering cells with percent.mt &lt; 10 ensures you retain only healthy cells with proper RNA integrity for downstream analyses.</span></span>
<span id="cb3-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-3" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">subset</span>(B11, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb3-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-4" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">subset</span>(B12, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb3-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-5" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">subset</span>(C1, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb3-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-6" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">subset</span>(C2, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb3-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-7" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">subset</span>(C3, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb3-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb3-8" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">subset</span>(C4, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span></code></pre></div>
<div class="figure">
<img src="images/subsetting_B11_B12.png" alt="Figure 1: Violin plots before and after subsetting of data &lt;10% mt. nFeature_RNA shows the distribution of the number of unique genes (features) detected in each cell. The majority of cells in unfiltered B11 appear to have between 1000 and 4000 unique features, with a small number of cells exceeding 6000 features. nCount_RNA shows the total number of RNA molecules (UMI counts) detected per cell. The distribution is more skewed compared to nFeature_RNA, with most cells having counts around 5000 to 15000, while a smaller number exceed 20,000 or 30,000. percent.mt  represents the percentage of mitochondrial gene expression in each cell. " width="920" />
<p class="caption">
(#fig:combined plot B11 and B12)Figure 1: Violin plots before and after subsetting of data &lt;10% mt. nFeature_RNA shows the distribution of the number of unique genes (features) detected in each cell. The majority of cells in unfiltered B11 appear to have between 1000 and 4000 unique features, with a small number of cells exceeding 6000 features. nCount_RNA shows the total number of RNA molecules (UMI counts) detected per cell. The distribution is more skewed compared to nFeature_RNA, with most cells having counts around 5000 to 15000, while a smaller number exceed 20,000 or 30,000. percent.mt represents the percentage of mitochondrial gene expression in each cell.
</p>
</div>
<p>A heavy concentration of dots at the lower end of the nFeature_RNA could indicate low-quality cells or droplets with ambient RNA. Cells with a high percentage of mitochondrial reads (e.g., above 20%) may indicate cell stress or damage. Most cells in this dataset appear to have mitochondrial percentages below 10%, with a few outliers exceeding 40%.</p>
<p>For further refining, the percent.mt has been set to &lt;3%</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-1" tabindex="-1"></a><span class="co"># Refine QC</span></span>
<span id="cb4-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-2" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">subset</span>(B11, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb4-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-3" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">subset</span>(B12, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb4-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-4" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">subset</span>(C1, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb4-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-5" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">subset</span>(C2, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb4-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-6" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">subset</span>(C3, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span>
<span id="cb4-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb4-7" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">subset</span>(C4, <span class="at">subset =</span> percent.mt <span class="sc">&lt;</span> <span class="dv">3</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">20000</span> )</span></code></pre></div>
<p>After filtering of the data, a cluster analysis is done. Doublets need to be removed and this step is necessary to perform for DoubletFinder.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-1" tabindex="-1"></a><span class="do">### Clustering analysis (necessary to perform for DoubletFinder)</span></span>
<span id="cb5-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-2" tabindex="-1"></a><span class="co"># Select the top 2000 most variable features</span></span>
<span id="cb5-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-3" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(B11,<span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb5-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-4" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(B12,<span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb5-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-5" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(C1,<span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb5-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-6" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(C2,<span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb5-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-7" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(C3,<span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb5-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-8" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">FindVariableFeatures</span>(C4,<span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb5-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-9" tabindex="-1"></a></span>
<span id="cb5-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-10" tabindex="-1"></a><span class="co"># Scaling RNA data, we only scale the variable features here for efficiency</span></span>
<span id="cb5-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-11" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">ScaleData</span>(B11, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(B11))</span>
<span id="cb5-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-12" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">ScaleData</span>(B12, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(B12))</span>
<span id="cb5-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-13" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">ScaleData</span>(C1, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C1))</span>
<span id="cb5-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-14" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">ScaleData</span>(C2, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C2))</span>
<span id="cb5-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-15" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">ScaleData</span>(C3, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C3))</span>
<span id="cb5-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-16" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">ScaleData</span>(C4, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C4))</span>
<span id="cb5-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-17" tabindex="-1"></a></span>
<span id="cb5-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-18" tabindex="-1"></a><span class="co"># Run PCA</span></span>
<span id="cb5-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-19" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">RunPCA</span>(B11, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(B11), <span class="at">verbose=</span>F)</span>
<span id="cb5-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-20" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">RunPCA</span>(B12, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(B12), <span class="at">verbose=</span>F)</span>
<span id="cb5-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-21" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">RunPCA</span>(C1, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C1), <span class="at">verbose=</span>F)</span>
<span id="cb5-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-22" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">RunPCA</span>(C2, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C2), <span class="at">verbose=</span>F)</span>
<span id="cb5-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-23" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">RunPCA</span>(C3, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C3), <span class="at">verbose=</span>F)</span>
<span id="cb5-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-24" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">RunPCA</span>(C4, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(C4), <span class="at">verbose=</span>F)</span>
<span id="cb5-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-25" tabindex="-1"></a></span>
<span id="cb5-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-26" tabindex="-1"></a><span class="co"># Clustering and dimensional reduction</span></span>
<span id="cb5-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-27" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">FindNeighbors</span>(B11, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-28" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">FindNeighbors</span>(B12, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-29" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">FindNeighbors</span>(C1, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-30" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">FindNeighbors</span>(C2, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-31"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-31" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">FindNeighbors</span>(C3, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-32"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-32" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">FindNeighbors</span>(C4, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-33"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-33" tabindex="-1"></a></span>
<span id="cb5-34"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-34" tabindex="-1"></a>B11 <span class="ot">=</span> <span class="fu">FindClusters</span>(B11, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-35"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-35" tabindex="-1"></a>B12 <span class="ot">=</span> <span class="fu">FindClusters</span>(B12, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-36"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-36" tabindex="-1"></a>C1 <span class="ot">=</span> <span class="fu">FindClusters</span>(C1, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-37"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-37" tabindex="-1"></a>C2 <span class="ot">=</span> <span class="fu">FindClusters</span>(C2, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-38"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-38" tabindex="-1"></a>C3 <span class="ot">=</span> <span class="fu">FindClusters</span>(C3, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-39"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-39" tabindex="-1"></a>C4 <span class="ot">=</span> <span class="fu">FindClusters</span>(C4, <span class="at">resolution =</span> <span class="fl">0.2</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-40"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-40" tabindex="-1"></a></span>
<span id="cb5-41"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-41" tabindex="-1"></a>B11<span class="ot">=</span><span class="fu">RunUMAP</span>(B11, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> F)</span>
<span id="cb5-42"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-42" tabindex="-1"></a>B12<span class="ot">=</span><span class="fu">RunUMAP</span>(B12, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> F)</span>
<span id="cb5-43"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-43" tabindex="-1"></a>C1<span class="ot">=</span><span class="fu">RunUMAP</span>(C1, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> F)</span>
<span id="cb5-44"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-44" tabindex="-1"></a>C2<span class="ot">=</span><span class="fu">RunUMAP</span>(C2, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> F)</span>
<span id="cb5-45"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-45" tabindex="-1"></a>C3<span class="ot">=</span><span class="fu">RunUMAP</span>(C3, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> F)</span>
<span id="cb5-46"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb5-46" tabindex="-1"></a>C4<span class="ot">=</span><span class="fu">RunUMAP</span>(C4, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> F)</span></code></pre></div>
<div class="figure">
<img src="images/UMAP_B11_B12.png" alt="Figure 2: UMAP visualisation of B11 and B12." width="918" />
<p class="caption">
(#fig:UMAP plot B11 and B12)Figure 2: UMAP visualisation of B11 and B12.
</p>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-1" tabindex="-1"></a><span class="co"># Add library metadata</span></span>
<span id="cb6-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-2" tabindex="-1"></a>B11[[<span class="st">&quot;Library&quot;</span>]]<span class="ot">=</span><span class="st">&quot;B11&quot;</span></span>
<span id="cb6-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-3" tabindex="-1"></a>B12[[<span class="st">&quot;Library&quot;</span>]]<span class="ot">=</span><span class="st">&quot;B12&quot;</span></span>
<span id="cb6-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-4" tabindex="-1"></a>C1[[<span class="st">&quot;Library&quot;</span>]]<span class="ot">=</span><span class="st">&quot;C1&quot;</span></span>
<span id="cb6-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-5" tabindex="-1"></a>C2[[<span class="st">&quot;Library&quot;</span>]]<span class="ot">=</span><span class="st">&quot;C2&quot;</span></span>
<span id="cb6-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-6" tabindex="-1"></a>C3[[<span class="st">&quot;Library&quot;</span>]]<span class="ot">=</span><span class="st">&quot;C3&quot;</span></span>
<span id="cb6-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb6-7" tabindex="-1"></a>C4[[<span class="st">&quot;Library&quot;</span>]]<span class="ot">=</span><span class="st">&quot;C4&quot;</span></span></code></pre></div>
<p>As previously mentioned, we have to remove the doublets from the dataset. In single-cell RNA sequencing, doublets and singlets are terms used to describe the state of captured cells during sequencing. A singlet refers to a single cell that has been successfully isolated into a droplet or well for sequencing. It represents the ideal scenario in scRNA-seq, where the transcriptome of only one individual cell is captured and analyzed. A doublet occurs when two or more cells are captured together in the same droplet or well during the isolation process. This leads to the combined RNA content of the captured cells being sequenced together, which can result in mixed or spurious gene expression profiles that don’t represent a single cell type. Doublets can arise due to technical limitations during cell partitioning in platforms like droplet-based methods (e.g., 10x Genomics Chromium).</p>
<p>Accurate distinction between singlets and doublets ensures that the scRNA-seq data faithfully represents the gene expression profiles of individual cells, which is crucial for downstream analyses like clustering, cell type identification, and differential gene expression analysis.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb7-1" tabindex="-1"></a><span class="do">### Doublet Finder </span><span class="al">###</span></span>
<span id="cb7-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb7-2" tabindex="-1"></a><span class="co"># pK Identification</span></span>
<span id="cb7-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb7-3" tabindex="-1"></a><span class="co"># let op VOOR runnen, duurt lang </span></span>
<span id="cb7-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb7-4" tabindex="-1"></a>sweep.res.list_B11 <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(B11, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb27-1" tabindex="-1"></a>sweep.res.list_B12 <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(B12, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb47-1" tabindex="-1"></a>sweep.res.list_C1 <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(C1, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb67-1" tabindex="-1"></a>sweep.res.list_C2 <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(C2, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb87-1" tabindex="-1"></a>sweep.res.list_C3 <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(C3, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb107-1" tabindex="-1"></a>sweep.res.list_C4 <span class="ot">&lt;-</span> <span class="fu">paramSweep</span>(C4, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating artificial doublets for pN = 5%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 10%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 15%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 20%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 25%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;
## [1] &quot;Creating artificial doublets for pN = 30%&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Defining neighborhoods...&quot;
## [1] &quot;Computing pANN across all pK...&quot;
## [1] &quot;pK = 0.005...&quot;
## [1] &quot;pK = 0.01...&quot;
## [1] &quot;pK = 0.02...&quot;
## [1] &quot;pK = 0.03...&quot;
## [1] &quot;pK = 0.04...&quot;
## [1] &quot;pK = 0.05...&quot;
## [1] &quot;pK = 0.06...&quot;
## [1] &quot;pK = 0.07...&quot;
## [1] &quot;pK = 0.08...&quot;
## [1] &quot;pK = 0.09...&quot;
## [1] &quot;pK = 0.1...&quot;
## [1] &quot;pK = 0.11...&quot;
## [1] &quot;pK = 0.12...&quot;
## [1] &quot;pK = 0.13...&quot;
## [1] &quot;pK = 0.14...&quot;
## [1] &quot;pK = 0.15...&quot;
## [1] &quot;pK = 0.16...&quot;
## [1] &quot;pK = 0.17...&quot;
## [1] &quot;pK = 0.18...&quot;
## [1] &quot;pK = 0.19...&quot;
## [1] &quot;pK = 0.2...&quot;
## [1] &quot;pK = 0.21...&quot;
## [1] &quot;pK = 0.22...&quot;
## [1] &quot;pK = 0.23...&quot;
## [1] &quot;pK = 0.24...&quot;
## [1] &quot;pK = 0.25...&quot;
## [1] &quot;pK = 0.26...&quot;
## [1] &quot;pK = 0.27...&quot;
## [1] &quot;pK = 0.28...&quot;
## [1] &quot;pK = 0.29...&quot;
## [1] &quot;pK = 0.3...&quot;</code></pre>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-1" tabindex="-1"></a><span class="co"># ^ tot hier</span></span>
<span id="cb127-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-2" tabindex="-1"></a></span>
<span id="cb127-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-3" tabindex="-1"></a>sweep.stats_B11 <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_B11, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb127-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-4" tabindex="-1"></a>sweep.stats_B12 <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_B12, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb127-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-5" tabindex="-1"></a>sweep.stats_C1 <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_C1, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb127-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-6" tabindex="-1"></a>sweep.stats_C2 <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_C2, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb127-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-7" tabindex="-1"></a>sweep.stats_C3 <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_C3, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb127-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-8" tabindex="-1"></a>sweep.stats_C4 <span class="ot">&lt;-</span> <span class="fu">summarizeSweep</span>(sweep.res.list_C4, <span class="at">GT =</span> <span class="cn">FALSE</span>)</span>
<span id="cb127-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-9" tabindex="-1"></a></span>
<span id="cb127-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb127-10" tabindex="-1"></a>bcmvn_B11 <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_B11)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb129-1" tabindex="-1"></a>bcmvn_B12 <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_B12)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb131-1" tabindex="-1"></a>bcmvn_C1 <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_C1)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb133-1" tabindex="-1"></a>bcmvn_C2 <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_C2)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb135-1" tabindex="-1"></a>bcmvn_C3 <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_C3)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb137-1" tabindex="-1"></a>bcmvn_C4 <span class="ot">&lt;-</span> <span class="fu">find.pK</span>(sweep.stats_C4)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p>pK specifies the fraction of k-nearest neighbors in the clustering analysis that should be artificial doublets. DoubletFinder uses a neighborhood-based approach to classify cells as singlets or doublets. The pK value determines the balance between real cells and artificial doublets in the space of k-nearest neighbors during this classification.
If pK is too low, there may not be enough artificial doublets to accurately model the distribution of real doublets. If pK is too high, it can overwhelm the clustering and lead to poor discrimination between real doublets and singlets.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-1" tabindex="-1"></a><span class="co"># Determine Optimal pK</span></span>
<span id="cb139-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-2" tabindex="-1"></a>bcmvn.max_B11 <span class="ot">&lt;-</span> bcmvn_B11[<span class="fu">which.max</span>(bcmvn_B11<span class="sc">$</span>BCmetric),]</span>
<span id="cb139-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-3" tabindex="-1"></a>bcmvn.max_B12 <span class="ot">&lt;-</span> bcmvn_B12[<span class="fu">which.max</span>(bcmvn_B12<span class="sc">$</span>BCmetric),]</span>
<span id="cb139-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-4" tabindex="-1"></a>bcmvn.max_C1 <span class="ot">&lt;-</span> bcmvn_C1[<span class="fu">which.max</span>(bcmvn_C1<span class="sc">$</span>BCmetric),]</span>
<span id="cb139-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-5" tabindex="-1"></a>bcmvn.max_C2 <span class="ot">&lt;-</span> bcmvn_C2[<span class="fu">which.max</span>(bcmvn_C2<span class="sc">$</span>BCmetric),]</span>
<span id="cb139-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-6" tabindex="-1"></a>bcmvn.max_C3 <span class="ot">&lt;-</span> bcmvn_C3[<span class="fu">which.max</span>(bcmvn_C3<span class="sc">$</span>BCmetric),]</span>
<span id="cb139-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-7" tabindex="-1"></a>bcmvn.max_C4 <span class="ot">&lt;-</span> bcmvn_C4[<span class="fu">which.max</span>(bcmvn_C4<span class="sc">$</span>BCmetric),]</span>
<span id="cb139-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-8" tabindex="-1"></a></span>
<span id="cb139-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-9" tabindex="-1"></a>optimal.pk_B11 <span class="ot">&lt;-</span> bcmvn.max_B11<span class="sc">$</span>pK</span>
<span id="cb139-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-10" tabindex="-1"></a>optimal.pk_B12 <span class="ot">&lt;-</span> bcmvn.max_B12<span class="sc">$</span>pK</span>
<span id="cb139-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-11" tabindex="-1"></a>optimal.pk_C1 <span class="ot">&lt;-</span> bcmvn.max_C1<span class="sc">$</span>pK</span>
<span id="cb139-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-12" tabindex="-1"></a>optimal.pk_C2 <span class="ot">&lt;-</span> bcmvn.max_C2<span class="sc">$</span>pK</span>
<span id="cb139-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-13" tabindex="-1"></a>optimal.pk_C3 <span class="ot">&lt;-</span> bcmvn.max_C3<span class="sc">$</span>pK</span>
<span id="cb139-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-14" tabindex="-1"></a>optimal.pk_C4 <span class="ot">&lt;-</span> bcmvn.max_C4<span class="sc">$</span>pK</span>
<span id="cb139-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-15" tabindex="-1"></a></span>
<span id="cb139-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-16" tabindex="-1"></a>optimal.pk_B11 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">levels</span>(optimal.pk_B11))[optimal.pk_B11]</span>
<span id="cb139-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-17" tabindex="-1"></a>optimal.pk_B12 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">levels</span>(optimal.pk_B12))[optimal.pk_B12]</span>
<span id="cb139-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-18" tabindex="-1"></a>optimal.pk_C1 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">levels</span>(optimal.pk_C1))[optimal.pk_C1]</span>
<span id="cb139-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-19" tabindex="-1"></a>optimal.pk_C2 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">levels</span>(optimal.pk_C2))[optimal.pk_C2]</span>
<span id="cb139-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-20" tabindex="-1"></a>optimal.pk_C3 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">levels</span>(optimal.pk_C3))[optimal.pk_C3]</span>
<span id="cb139-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-21" tabindex="-1"></a>optimal.pk_C4 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">levels</span>(optimal.pk_C4))[optimal.pk_C4]</span>
<span id="cb139-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-22" tabindex="-1"></a></span>
<span id="cb139-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-23" tabindex="-1"></a>optimal_pk_tibble <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb139-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-24" tabindex="-1"></a>  <span class="at">Sample =</span> <span class="fu">c</span>(<span class="st">&quot;B11&quot;</span>, <span class="st">&quot;B12&quot;</span>, <span class="st">&quot;C1&quot;</span>, <span class="st">&quot;C2&quot;</span>, <span class="st">&quot;C3&quot;</span>, <span class="st">&quot;C4&quot;</span>),</span>
<span id="cb139-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-25" tabindex="-1"></a>  <span class="at">optimal_pK =</span> <span class="fu">c</span>(optimal.pk_B11, optimal.pk_B12, optimal.pk_C1, </span>
<span id="cb139-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-26" tabindex="-1"></a>                 optimal.pk_C2, optimal.pk_C3, optimal.pk_C4)</span>
<span id="cb139-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-27" tabindex="-1"></a>)</span>
<span id="cb139-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-28" tabindex="-1"></a></span>
<span id="cb139-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb139-29" tabindex="-1"></a>optimal_pk_tibble</span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   Sample optimal_pK
##   &lt;chr&gt;       &lt;dbl&gt;
## 1 B11         0.01 
## 2 B12         0.005
## 3 C1          0.005
## 4 C2          0.1  
## 5 C3          0.005
## 6 C4          0.005</code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-1" tabindex="-1"></a><span class="co"># Homotypic Doublet Proportion Estimate </span></span>
<span id="cb141-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-2" tabindex="-1"></a>annotations_B11<span class="ot">=</span>B11<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb141-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-3" tabindex="-1"></a>annotations_B12<span class="ot">=</span>B12<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb141-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-4" tabindex="-1"></a>annotations_C1<span class="ot">=</span>C1<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb141-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-5" tabindex="-1"></a>annotations_C2<span class="ot">=</span>C2<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb141-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-6" tabindex="-1"></a>annotations_C3<span class="ot">=</span>C3<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb141-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-7" tabindex="-1"></a>annotations_C4<span class="ot">=</span>C4<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters</span>
<span id="cb141-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-8" tabindex="-1"></a></span>
<span id="cb141-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-9" tabindex="-1"></a>homotypic.prop_B11 <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations_B11)     </span>
<span id="cb141-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-10" tabindex="-1"></a>homotypic.prop_B12 <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations_B12)           </span>
<span id="cb141-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-11" tabindex="-1"></a>homotypic.prop_C1 <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations_C1)           </span>
<span id="cb141-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-12" tabindex="-1"></a>homotypic.prop_C2 <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations_C2)           </span>
<span id="cb141-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-13" tabindex="-1"></a>homotypic.prop_C3 <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations_C3)           </span>
<span id="cb141-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-14" tabindex="-1"></a>homotypic.prop_C4 <span class="ot">&lt;-</span> <span class="fu">modelHomotypic</span>(annotations_C4)           </span>
<span id="cb141-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-15" tabindex="-1"></a></span>
<span id="cb141-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-16" tabindex="-1"></a>nExp_poi_B11 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.15</span><span class="sc">*</span><span class="fu">nrow</span>(B11<span class="sc">@</span>meta.data))  <span class="do">## Assuming 15% doublet formation rate </span></span>
<span id="cb141-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-17" tabindex="-1"></a>nExp_poi_B12 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.15</span><span class="sc">*</span><span class="fu">nrow</span>(B12<span class="sc">@</span>meta.data))  </span>
<span id="cb141-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-18" tabindex="-1"></a>nExp_poi_C1 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.15</span><span class="sc">*</span><span class="fu">nrow</span>(C1<span class="sc">@</span>meta.data))  </span>
<span id="cb141-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-19" tabindex="-1"></a>nExp_poi_C2 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.15</span><span class="sc">*</span><span class="fu">nrow</span>(C2<span class="sc">@</span>meta.data))  </span>
<span id="cb141-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-20" tabindex="-1"></a>nExp_poi_C3 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.15</span><span class="sc">*</span><span class="fu">nrow</span>(C3<span class="sc">@</span>meta.data))  </span>
<span id="cb141-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-21" tabindex="-1"></a>nExp_poi_C4 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.15</span><span class="sc">*</span><span class="fu">nrow</span>(C4<span class="sc">@</span>meta.data))  </span>
<span id="cb141-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-22" tabindex="-1"></a></span>
<span id="cb141-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-23" tabindex="-1"></a>nExp_poi.adj_B11 <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi_B11<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>homotypic.prop_B11))</span>
<span id="cb141-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-24" tabindex="-1"></a>nExp_poi.adj_B12 <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi_B12<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>homotypic.prop_B12))</span>
<span id="cb141-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-25" tabindex="-1"></a>nExp_poi.adj_C1 <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi_C1<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>homotypic.prop_C1))</span>
<span id="cb141-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-26" tabindex="-1"></a>nExp_poi.adj_C2 <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi_C2<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>homotypic.prop_C2))</span>
<span id="cb141-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-27" tabindex="-1"></a>nExp_poi.adj_C3 <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi_C3<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>homotypic.prop_C3))</span>
<span id="cb141-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-28" tabindex="-1"></a>nExp_poi.adj_C4 <span class="ot">&lt;-</span> <span class="fu">round</span>(nExp_poi_C4<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>homotypic.prop_C4))</span>
<span id="cb141-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-29" tabindex="-1"></a></span>
<span id="cb141-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-30" tabindex="-1"></a><span class="co"># Run Doublet Finder</span></span>
<span id="cb141-31"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb141-31" tabindex="-1"></a>B11 <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(B11, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> optimal.pk_B11, <span class="at">nExp =</span> nExp_poi.adj_B11, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating 2342 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb146-1" tabindex="-1"></a>B12 <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(B12, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> optimal.pk_B12, <span class="at">nExp =</span> nExp_poi.adj_B12, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating 3072 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb151-1" tabindex="-1"></a>C1 <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(C1, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> optimal.pk_C1, <span class="at">nExp =</span> nExp_poi.adj_C1, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating 2073 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb156-1" tabindex="-1"></a>C2 <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(C2, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> optimal.pk_C2, <span class="at">nExp =</span> nExp_poi.adj_C2, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating 3047 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb161-1" tabindex="-1"></a>C3 <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(C3, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> optimal.pk_C3, <span class="at">nExp =</span> nExp_poi.adj_C3, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating 2444 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb166-1" tabindex="-1"></a>C4 <span class="ot">&lt;-</span> <span class="fu">doubletFinder</span>(C4, <span class="at">PCs =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pN =</span> <span class="fl">0.25</span>, <span class="at">pK =</span> optimal.pk_C4, <span class="at">nExp =</span> nExp_poi.adj_C4, <span class="at">reuse.pANN =</span> <span class="cn">FALSE</span>, <span class="at">sct =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Creating 3003 artificial doublets...&quot;
## [1] &quot;Creating Seurat object...&quot;
## [1] &quot;Normalizing Seurat object...&quot;</code></pre>
<pre><code>## [1] &quot;Finding variable genes...&quot;</code></pre>
<pre><code>## [1] &quot;Scaling data...&quot;</code></pre>
<pre><code>## [1] &quot;Running PCA...&quot;
## [1] &quot;Calculating PC distance matrix...&quot;
## [1] &quot;Computing pANN...&quot;
## [1] &quot;Classifying doublets..&quot;</code></pre>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb171-1" tabindex="-1"></a><span class="co">#Do this for B11 to C4, to be able to check and adjust the DF.classification number. </span></span>
<span id="cb171-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb171-2" tabindex="-1"></a><span class="fu">head</span>(B11<span class="sc">@</span>meta.data)</span></code></pre></div>
<pre><code>##                       orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACCCAAGCACTTTG-1 SeuratProject       6145         2814  2.1480879
## AAACCCAAGCGCTTCG-1 SeuratProject       3112         1729  1.0282776
## AAACCCAAGTGCCCGT-1 SeuratProject       5922         2349  0.2364066
## AAACCCACAGTTGAAA-1 SeuratProject       9442         2988  0.1059098
## AAACCCAGTACGATCT-1 SeuratProject       5218         2126  0.1916443
## AAACCCATCGCTACGG-1 SeuratProject      14109         3831  0.0283507
##                    RNA_snn_res.0.2 seurat_clusters Library pANN_0.25_0.01_921
## AAACCCAAGCACTTTG-1               3               3     B11         0.14893617
## AAACCCAAGCGCTTCG-1               9               9     B11         0.06382979
## AAACCCAAGTGCCCGT-1               1               1     B11         0.04255319
## AAACCCACAGTTGAAA-1               0               0     B11         0.15957447
## AAACCCAGTACGATCT-1               7               7     B11         0.17021277
## AAACCCATCGCTACGG-1               4               4     B11         0.08510638
##                    DF.classifications_0.25_0.01_921
## AAACCCAAGCACTTTG-1                          Singlet
## AAACCCAAGCGCTTCG-1                          Singlet
## AAACCCAAGTGCCCGT-1                          Singlet
## AAACCCACAGTTGAAA-1                          Singlet
## AAACCCAGTACGATCT-1                          Singlet
## AAACCCATCGCTACGG-1                          Singlet</code></pre>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-1" tabindex="-1"></a><span class="co">#Identify the metadata column corresponding to the doublet/singlet identity identified by DoubletFinder, here &quot;DF.classifications_0.25_0.01_921&quot;, this must be adjusted for each variable</span></span>
<span id="cb173-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-2" tabindex="-1"></a><span class="fu">Idents</span>(B11)<span class="ot">=</span><span class="st">&quot;DF.classifications_0.25_0.01_921&quot;</span></span>
<span id="cb173-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-3" tabindex="-1"></a><span class="fu">Idents</span>(B12)<span class="ot">=</span><span class="st">&quot;DF.classifications_0.25_0.005_1146&quot;</span></span>
<span id="cb173-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-4" tabindex="-1"></a><span class="fu">Idents</span>(C1)<span class="ot">=</span><span class="st">&quot;DF.classifications_0.25_0.005_774&quot;</span></span>
<span id="cb173-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-5" tabindex="-1"></a><span class="fu">Idents</span>(C2)<span class="ot">=</span><span class="st">&quot;DF.classifications_0.25_0.1_1130&quot;</span></span>
<span id="cb173-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-6" tabindex="-1"></a><span class="fu">Idents</span>(C3)<span class="ot">=</span><span class="st">&quot;DF.classifications_0.25_0.005_947&quot;</span></span>
<span id="cb173-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb173-7" tabindex="-1"></a><span class="fu">Idents</span>(C4)<span class="ot">=</span><span class="st">&quot;DF.classifications_0.25_0.005_1121&quot;</span></span></code></pre></div>
<div class="figure">
<img src="images/doublets_B11_B12.png" alt="Figure 3: UMAP visualisation of B11 and B12." width="850" />
<p class="caption">
(#fig:doublet visualisation B11 B12)Figure 3: UMAP visualisation of B11 and B12.
</p>
</div>
<p>Nu dat we de doublets hebben geidentificeerd en gevisualiseerd, kunnen we de data opnieuw subsetten om enkel de singlets te behouden.</p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb174-1" tabindex="-1"></a>B11_DF<span class="ot">=</span><span class="fu">subset</span>(B11, <span class="at">idents =</span> <span class="st">&quot;Singlet&quot;</span>)</span>
<span id="cb174-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb174-2" tabindex="-1"></a>B12_DF<span class="ot">=</span><span class="fu">subset</span>(B12, <span class="at">idents =</span> <span class="st">&quot;Singlet&quot;</span>)</span>
<span id="cb174-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb174-3" tabindex="-1"></a>C1_DF<span class="ot">=</span><span class="fu">subset</span>(C1, <span class="at">idents =</span> <span class="st">&quot;Singlet&quot;</span>)</span>
<span id="cb174-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb174-4" tabindex="-1"></a>C2_DF<span class="ot">=</span><span class="fu">subset</span>(C2, <span class="at">idents =</span> <span class="st">&quot;Singlet&quot;</span>)</span>
<span id="cb174-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb174-5" tabindex="-1"></a>C3_DF<span class="ot">=</span><span class="fu">subset</span>(C3, <span class="at">idents =</span> <span class="st">&quot;Singlet&quot;</span>)</span>
<span id="cb174-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb174-6" tabindex="-1"></a>C4_DF<span class="ot">=</span><span class="fu">subset</span>(C4, <span class="at">idents =</span> <span class="st">&quot;Singlet&quot;</span>)</span></code></pre></div>
<p>Nu hebben we voor elk “sample” apart de data verwerkt. Om verdere analyses uit te kunnen voeren, moeten deze eerst samengevoegd, “gepoold”, worden.</p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-1" tabindex="-1"></a><span class="do">########### Pool the 6 QC&#39;d libraries and proceed to analysis ###############</span></span>
<span id="cb175-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-2" tabindex="-1"></a></span>
<span id="cb175-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-3" tabindex="-1"></a><span class="co"># Add sample info</span></span>
<span id="cb175-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-4" tabindex="-1"></a>B11_DF[[<span class="st">&quot;Sample&quot;</span>]]<span class="ot">=</span><span class="st">&quot;WT_1&quot;</span></span>
<span id="cb175-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-5" tabindex="-1"></a>B12_DF[[<span class="st">&quot;Sample&quot;</span>]]<span class="ot">=</span><span class="st">&quot;WT_2&quot;</span></span>
<span id="cb175-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-6" tabindex="-1"></a>C1_DF[[<span class="st">&quot;Sample&quot;</span>]]<span class="ot">=</span><span class="st">&quot;WT_3&quot;</span></span>
<span id="cb175-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-7" tabindex="-1"></a>C2_DF[[<span class="st">&quot;Sample&quot;</span>]]<span class="ot">=</span><span class="st">&quot;KO_1&quot;</span></span>
<span id="cb175-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-8" tabindex="-1"></a>C3_DF[[<span class="st">&quot;Sample&quot;</span>]]<span class="ot">=</span><span class="st">&quot;KO_2&quot;</span></span>
<span id="cb175-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-9" tabindex="-1"></a>C4_DF[[<span class="st">&quot;Sample&quot;</span>]]<span class="ot">=</span><span class="st">&quot;KO_3&quot;</span></span>
<span id="cb175-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-10" tabindex="-1"></a></span>
<span id="cb175-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-11" tabindex="-1"></a><span class="co"># Add experimental condition info</span></span>
<span id="cb175-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-12" tabindex="-1"></a>B11_DF[[<span class="st">&quot;Condition&quot;</span>]]<span class="ot">=</span><span class="st">&quot;WT&quot;</span></span>
<span id="cb175-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-13" tabindex="-1"></a>B12_DF[[<span class="st">&quot;Condition&quot;</span>]]<span class="ot">=</span><span class="st">&quot;WT&quot;</span></span>
<span id="cb175-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-14" tabindex="-1"></a>C1_DF[[<span class="st">&quot;Condition&quot;</span>]]<span class="ot">=</span><span class="st">&quot;WT&quot;</span></span>
<span id="cb175-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-15" tabindex="-1"></a>C2_DF[[<span class="st">&quot;Condition&quot;</span>]]<span class="ot">=</span><span class="st">&quot;KO&quot;</span></span>
<span id="cb175-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-16" tabindex="-1"></a>C3_DF[[<span class="st">&quot;Condition&quot;</span>]]<span class="ot">=</span><span class="st">&quot;KO&quot;</span></span>
<span id="cb175-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-17" tabindex="-1"></a>C4_DF[[<span class="st">&quot;Condition&quot;</span>]]<span class="ot">=</span><span class="st">&quot;KO&quot;</span></span>
<span id="cb175-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-18" tabindex="-1"></a></span>
<span id="cb175-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-19" tabindex="-1"></a><span class="co"># Merge data</span></span>
<span id="cb175-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-20" tabindex="-1"></a>Trem2_nuc<span class="ot">=</span><span class="fu">merge</span>(<span class="at">x=</span>B11_DF, <span class="at">y=</span><span class="fu">list</span>(B12_DF, C1_DF, C2_DF, C3_DF, C4_DF))</span>
<span id="cb175-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-21" tabindex="-1"></a></span>
<span id="cb175-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-22" tabindex="-1"></a><span class="co"># Dimensional Reduction and clustering analysis</span></span>
<span id="cb175-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-23" tabindex="-1"></a><span class="fu">DefaultAssay</span>(Trem2_nuc)<span class="ot">&lt;-</span> <span class="st">&quot;RNA&quot;</span></span>
<span id="cb175-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-24" tabindex="-1"></a>Trem2_nuc <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(Trem2_nuc, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb175-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-25" tabindex="-1"></a>Trem2_nuc <span class="ot">&lt;-</span><span class="fu">FindVariableFeatures</span>(Trem2_nuc, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb175-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-26" tabindex="-1"></a>Trem2_nuc <span class="ot">&lt;-</span><span class="fu">ScaleData</span>(Trem2_nuc,<span class="at">verbose =</span> <span class="cn">FALSE</span>) </span>
<span id="cb175-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-27" tabindex="-1"></a>Trem2_nuc <span class="ot">&lt;-</span><span class="fu">RunPCA</span>(Trem2_nuc, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb175-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-28" tabindex="-1"></a>Trem2_nuc <span class="ot">=</span> <span class="fu">RunUMAP</span>(Trem2_nuc, <span class="at">dims=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>,<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb175-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-29" tabindex="-1"></a>Trem2_nuc<span class="ot">=</span><span class="fu">FindNeighbors</span>(Trem2_nuc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb175-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb175-30" tabindex="-1"></a>Trem2_nuc<span class="ot">=</span><span class="fu">FindClusters</span>(Trem2_nuc, <span class="at">resolution =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 41906
## Number of edges: 1482810
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9756
## Number of communities: 13
## Elapsed time: 7 seconds</code></pre>
<div class="figure">
<img src="_main_files/figure-html/total%20data%20vis-1.png" alt="Figure 4: UMAP of combined datasets. 10 cluster were identified using FindClusters." width="672" />
<p class="caption">
(#fig:total data vis)Figure 4: UMAP of combined datasets. 10 cluster were identified using FindClusters.
</p>
</div>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-1" tabindex="-1"></a><span class="co"># When running this, an error may pop up that there isn&#39;t sufficient storage. Use rm(). </span></span>
<span id="cb177-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-2" tabindex="-1"></a>Trem2_nuc <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(Trem2_nuc) </span>
<span id="cb177-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-3" tabindex="-1"></a><span class="co">#Markers_Lineage=FindAllMarkers(Trem2_nuc, max.cells.per.ident = 250, logfc.threshold = 0.3, only.pos = T)</span></span>
<span id="cb177-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-4" tabindex="-1"></a><span class="co">#top20 &lt;- Markers_Lineage %&gt;% </span></span>
<span id="cb177-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-5" tabindex="-1"></a><span class="co">#  group_by(cluster) %&gt;% </span></span>
<span id="cb177-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-6" tabindex="-1"></a><span class="co">#  top_n(n =20 , wt = avg_log2FC)</span></span>
<span id="cb177-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb177-7" tabindex="-1"></a><span class="co">#top20 </span></span></code></pre></div>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-1" tabindex="-1"></a><span class="co"># Annotate</span></span>
<span id="cb178-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-2" tabindex="-1"></a>new.cluster.ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;VSMC&quot;</span>,</span>
<span id="cb178-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-3" tabindex="-1"></a>                     <span class="st">&quot;EC_1&quot;</span>,</span>
<span id="cb178-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-4" tabindex="-1"></a>                     <span class="st">&quot;Adipocyte&quot;</span>,</span>
<span id="cb178-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-5" tabindex="-1"></a>                     <span class="st">&quot;Mac/Myelo/APC&quot;</span>,</span>
<span id="cb178-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-6" tabindex="-1"></a>                     <span class="st">&quot;Fibro&quot;</span>,</span>
<span id="cb178-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-7" tabindex="-1"></a>                     <span class="st">&quot;EC_2&quot;</span>,</span>
<span id="cb178-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-8" tabindex="-1"></a>                     <span class="st">&quot;Glial cells&quot;</span>,</span>
<span id="cb178-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-9" tabindex="-1"></a>                     <span class="st">&quot;T_Cells&quot;</span>,</span>
<span id="cb178-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-10" tabindex="-1"></a>                     <span class="st">&quot;B_cells&quot;</span>,</span>
<span id="cb178-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-11" tabindex="-1"></a>                     <span class="st">&quot;RBC&quot;</span>,</span>
<span id="cb178-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-12" tabindex="-1"></a>                     <span class="st">&quot;Extra_cluster&quot;</span>) <span class="co">#Verschil met originele data. Dit is zelf toegevoegd. </span></span>
<span id="cb178-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-13" tabindex="-1"></a><span class="fu">names</span>(new.cluster.ids) <span class="ot">&lt;-</span> <span class="fu">levels</span>(Trem2_nuc)</span>
<span id="cb178-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-14" tabindex="-1"></a>Trem2_nuc<span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(Trem2_nuc, new.cluster.ids)</span>
<span id="cb178-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-15" tabindex="-1"></a><span class="co">#Trem2_nuc &lt;- StashIdent(object = Trem2_nuc, save.name = &quot;Lineage&quot;)</span></span>
<span id="cb178-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-16" tabindex="-1"></a>Trem2_nuc[[<span class="st">&quot;Lineage&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">Idents</span>(Trem2_nuc)</span>
<span id="cb178-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-17" tabindex="-1"></a></span>
<span id="cb178-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-18" tabindex="-1"></a><span class="fu">levels</span>(Trem2_nuc)<span class="ot">=</span><span class="fu">c</span>(</span>
<span id="cb178-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-19" tabindex="-1"></a>  <span class="st">&quot;EC_1&quot;</span>,</span>
<span id="cb178-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-20" tabindex="-1"></a>  <span class="st">&quot;EC_2&quot;</span>,</span>
<span id="cb178-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-21" tabindex="-1"></a>  <span class="st">&quot;VSMC&quot;</span>,</span>
<span id="cb178-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-22" tabindex="-1"></a>  <span class="st">&quot;Fibro&quot;</span>,</span>
<span id="cb178-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-23" tabindex="-1"></a>  <span class="st">&quot;Mac/Myelo/APC&quot;</span>,</span>
<span id="cb178-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-24" tabindex="-1"></a>  <span class="st">&quot;Adipocyte&quot;</span>,</span>
<span id="cb178-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-25" tabindex="-1"></a>  <span class="st">&quot;Glial cells&quot;</span>,</span>
<span id="cb178-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-26" tabindex="-1"></a>  <span class="st">&quot;T_Cells&quot;</span>,</span>
<span id="cb178-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-27" tabindex="-1"></a>  <span class="st">&quot;B_cells&quot;</span>,</span>
<span id="cb178-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-28" tabindex="-1"></a>  <span class="st">&quot;RBC&quot;</span>,</span>
<span id="cb178-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-29" tabindex="-1"></a>  <span class="st">&quot;Extra_cluster&quot;</span>)</span>
<span id="cb178-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-30" tabindex="-1"></a></span>
<span id="cb178-31"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-31" tabindex="-1"></a><span class="co"># Rename to ensure alphabetical order</span></span>
<span id="cb178-32"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-32" tabindex="-1"></a>new.cluster.ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;a_EC_1&quot;</span>,</span>
<span id="cb178-33"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-33" tabindex="-1"></a>                     <span class="st">&quot;b_EC_2&quot;</span>,</span>
<span id="cb178-34"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-34" tabindex="-1"></a>                     <span class="st">&quot;c_VSMC&quot;</span>,</span>
<span id="cb178-35"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-35" tabindex="-1"></a>                     <span class="st">&quot;d_Fibro&quot;</span>,</span>
<span id="cb178-36"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-36" tabindex="-1"></a>                     <span class="st">&quot;e_Mac/Myelo/APC&quot;</span>,</span>
<span id="cb178-37"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-37" tabindex="-1"></a>                     <span class="st">&quot;f_Adipocyte&quot;</span>,</span>
<span id="cb178-38"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-38" tabindex="-1"></a>                     <span class="st">&quot;g_Glial cells&quot;</span>,</span>
<span id="cb178-39"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-39" tabindex="-1"></a>                     <span class="st">&quot;h_T_Cells&quot;</span>,</span>
<span id="cb178-40"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-40" tabindex="-1"></a>                     <span class="st">&quot;i_B_cells&quot;</span>,</span>
<span id="cb178-41"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-41" tabindex="-1"></a>                     <span class="st">&quot;j_RBC&quot;</span>,</span>
<span id="cb178-42"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-42" tabindex="-1"></a>                     <span class="st">&quot;k_Extra_cluster&quot;</span>)</span>
<span id="cb178-43"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-43" tabindex="-1"></a><span class="fu">names</span>(new.cluster.ids) <span class="ot">&lt;-</span> <span class="fu">levels</span>(Trem2_nuc)</span>
<span id="cb178-44"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-44" tabindex="-1"></a>Trem2_nuc<span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(Trem2_nuc, new.cluster.ids)</span>
<span id="cb178-45"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-45" tabindex="-1"></a>Trem2_nuc[[<span class="st">&quot;Lineage&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">Idents</span>(Trem2_nuc)</span>
<span id="cb178-46"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-46" tabindex="-1"></a></span>
<span id="cb178-47"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-47" tabindex="-1"></a><span class="co"># Extended data Figure 6b</span></span>
<span id="cb178-48"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb178-48" tabindex="-1"></a><span class="co">#UMAPPlot(Trem2_nuc)+scale_color_brewer(palette = &quot;Paired&quot;)</span></span></code></pre></div>
<div class="figure">
<img src="images/fig_6b.png" alt="Figure 5:  UMAP visualization of aortic single-nucleus transcriptomes from Ldlr−/−Trem2+/+ (n = 3 females) and Ldlr−/−Trem2−/− (n = 3 females) mice after 10 weeks of HFD feeding (MPC:mononuclear phagocytes, VSMC: vascular smooth muscle cells)" width="1164" />
<p class="caption">
(#fig:comparing figuur 6b)Figure 5: UMAP visualization of aortic single-nucleus transcriptomes from Ldlr−/−Trem2+/+ (n = 3 females) and Ldlr−/−Trem2−/− (n = 3 females) mice after 10 weeks of HFD feeding (MPC:mononuclear phagocytes, VSMC: vascular smooth muscle cells)
</p>
</div>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb179-1" tabindex="-1"></a><span class="co"># Extended data Figure 6c</span></span>
<span id="cb179-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb179-2" tabindex="-1"></a><span class="fu">DotPlot</span>(Trem2_nuc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;Pecam1&quot;</span>, <span class="st">&quot;Tek&quot;</span>, <span class="st">&quot;Emcn&quot;</span>, <span class="st">&quot;Fli1&quot;</span>, <span class="st">&quot;Myh11&quot;</span>, <span class="st">&quot;Acta2&quot;</span>, <span class="st">&quot;Dcn&quot;</span>,<span class="st">&quot;Col14a1&quot;</span>, <span class="st">&quot;Sirpa&quot;</span>, <span class="st">&quot;Lgals3&quot;</span>, <span class="st">&quot;Plin1&quot;</span>, </span>
<span id="cb179-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb179-3" tabindex="-1"></a>                                <span class="st">&quot;Slc36a2&quot;</span>, <span class="st">&quot;Ncam1&quot;</span>, <span class="st">&quot;Grid2&quot;</span>, <span class="st">&quot;Lck&quot;</span>, <span class="st">&quot;Il7r&quot;</span>,</span>
<span id="cb179-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb179-4" tabindex="-1"></a>                                <span class="st">&quot;Ighm&quot;</span>, <span class="st">&quot;Ighd&quot;</span>, <span class="st">&quot;Hbb-bt&quot;</span>, <span class="st">&quot;Hba-a2&quot;</span>),<span class="at">cols=</span><span class="st">&quot;RdYlBu&quot;</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<div class="figure">
<img src="images/fig_6c.png" alt="Figure 6: expression of marker genes defining cell lineages" width="1141" />
<p class="caption">
(#fig:comparing 6c)Figure 6: expression of marker genes defining cell lineages
</p>
</div>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb180-1" tabindex="-1"></a><span class="co"># Extended data Figure 8</span></span>
<span id="cb180-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb180-2" tabindex="-1"></a><span class="co">#VlnPlot(Trem2_nuc2, &quot;Cd47&quot;, pt.size = 0, split.by = &quot;Condition&quot;)+scale_fill_manual(values = c(&quot;coral&quot;, &quot;turquoise4&quot;))</span></span>
<span id="cb180-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb180-3" tabindex="-1"></a></span>
<span id="cb180-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb180-4" tabindex="-1"></a>Trem2_nuc2<span class="ot">&lt;-</span> Trem2_nuc<span class="sc">$</span>Condition <span class="ot">&lt;-</span> <span class="fu">factor</span>(Trem2_nuc<span class="sc">$</span>Condition, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;KO&quot;</span>))</span>
<span id="cb180-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb180-5" tabindex="-1"></a><span class="fu">VlnPlot</span>(Trem2_nuc, <span class="st">&quot;Cd47&quot;</span>, <span class="at">pt.size =</span> <span class="dv">0</span>, <span class="at">split.by =</span> <span class="st">&quot;Condition&quot;</span>) <span class="sc">+</span></span>
<span id="cb180-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb180-6" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;turquoise4&quot;</span>, <span class="st">&quot;coral&quot;</span>))</span></code></pre></div>
<div class="figure">
<img src="images/fig_8.png" alt="Figure 7: Normalized Cd47 transcript expression in single-nucleus RNA seq data in aortic cell populations identified in Supplementary Figure 6, split according to genotype." width="1341" />
<p class="caption">
(#fig:vergelijking figure 8)Figure 7: Normalized Cd47 transcript expression in single-nucleus RNA seq data in aortic cell populations identified in Supplementary Figure 6, split according to genotype.
</p>
</div>
<p>Higher values suggest stronger expression of the Cd47 gene.
Orange violins represent KO (knockout) cells, where the Cd47 gene has been disrupted.
Green violins represent WT (wild type) cells, with normal Cd47 function.
The width of each violin represents the density of cells at a given expression level.
Wider areas suggest that more cells have that specific expression level.</p>
<p>In some cell types (e.g., e_Mac/Myelo/APC, f_Adipocyte), the expression levels are visibly different between KO and WT: e_Mac/Myelo/APC: Higher expression in WT compared to KO; f_Adipocyte: Slightly higher expression in KO compared to WT. These differences suggest that Cd47 is differentially regulated in these populations.</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-1" tabindex="-1"></a>Trem2_nuc_mac<span class="ot">=</span><span class="fu">subset</span>(Trem2_nuc, <span class="at">idents=</span><span class="fu">c</span>(<span class="st">&quot;e_Mac/Myelo/APC&quot;</span>))</span>
<span id="cb181-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-2" tabindex="-1"></a></span>
<span id="cb181-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-3" tabindex="-1"></a></span>
<span id="cb181-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-4" tabindex="-1"></a>Trem2_nuc_mac <span class="ot">&lt;-</span><span class="fu">FindVariableFeatures</span>(Trem2_nuc_mac, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb181-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-5" tabindex="-1"></a>Trem2_nuc_mac <span class="ot">&lt;-</span><span class="fu">ScaleData</span>(Trem2_nuc_mac,<span class="at">verbose =</span> <span class="cn">FALSE</span>) </span>
<span id="cb181-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-6" tabindex="-1"></a>Trem2_nuc_mac <span class="ot">&lt;-</span><span class="fu">RunPCA</span>(Trem2_nuc_mac, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb181-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-7" tabindex="-1"></a>Trem2_nuc_mac <span class="ot">=</span> <span class="fu">RunUMAP</span>(Trem2_nuc_mac, <span class="at">dims=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb181-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-8" tabindex="-1"></a>Trem2_nuc_mac<span class="ot">=</span><span class="fu">FindNeighbors</span>(Trem2_nuc_mac, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb181-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-9" tabindex="-1"></a><span class="fu">suppressMessages</span>({</span>
<span id="cb181-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-10" tabindex="-1"></a>Trem2_nuc_mac<span class="ot">=</span><span class="fu">FindClusters</span>(Trem2_nuc_mac, <span class="at">resolution =</span> <span class="fl">0.3</span>)</span>
<span id="cb181-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb181-11" tabindex="-1"></a>})</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 4534
## Number of edges: 155770
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8850
## Number of communities: 9
## Elapsed time: 0 seconds</code></pre>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb183-1" tabindex="-1"></a><span class="co">#UMAPPlot(Trem2_nuc_mac, label=T)</span></span></code></pre></div>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb184-1" tabindex="-1"></a><span class="do">##########deze duurt ook lang man</span></span>
<span id="cb184-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb184-2" tabindex="-1"></a>Markers_Mac<span class="ot">=</span><span class="fu">FindAllMarkers</span>(Trem2_nuc_mac, <span class="at">only.pos =</span> T)</span>
<span id="cb184-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb184-3" tabindex="-1"></a><span class="co">#top20_mac &lt;- Markers_Mac %&gt;% group_by(cluster) %&gt;% top_n(n = 20 , wt = avg_log2FC)</span></span>
<span id="cb184-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb184-4" tabindex="-1"></a><span class="co">#top20_mac</span></span></code></pre></div>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-1" tabindex="-1"></a><span class="co"># Annotate</span></span>
<span id="cb185-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-2" tabindex="-1"></a>new.cluster.ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Foamy Mac&quot;</span>,</span>
<span id="cb185-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-3" tabindex="-1"></a>                     <span class="st">&quot;Mac_Tgfbr1&quot;</span>,</span>
<span id="cb185-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-4" tabindex="-1"></a>                     <span class="st">&quot;Resident_Mac&quot;</span>,</span>
<span id="cb185-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-5" tabindex="-1"></a>                     <span class="st">&quot;low_quality&quot;</span>,</span>
<span id="cb185-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-6" tabindex="-1"></a>                     <span class="st">&quot;Mono&quot;</span>,</span>
<span id="cb185-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-7" tabindex="-1"></a>                     <span class="st">&quot;Cavity_Mac&quot;</span>,</span>
<span id="cb185-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-8" tabindex="-1"></a>                     <span class="st">&quot;Mac_Cd209+&quot;</span>,</span>
<span id="cb185-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-9" tabindex="-1"></a>                     <span class="st">&quot;nd&quot;</span>,</span>
<span id="cb185-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-10" tabindex="-1"></a>                     <span class="st">&quot;cDC2&quot;</span>,</span>
<span id="cb185-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-11" tabindex="-1"></a>                     <span class="st">&quot;cDC1&quot;</span>,</span>
<span id="cb185-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-12" tabindex="-1"></a>                     <span class="st">&quot;Mature_DC&quot;</span>,</span>
<span id="cb185-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-13" tabindex="-1"></a>                     <span class="st">&quot;Myelo_Padi4+&quot;</span>)</span>
<span id="cb185-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-14" tabindex="-1"></a><span class="fu">names</span>(new.cluster.ids) <span class="ot">&lt;-</span> <span class="fu">levels</span>(Trem2_nuc_mac)</span>
<span id="cb185-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-15" tabindex="-1"></a>Trem2_nuc_mac<span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(Trem2_nuc_mac, new.cluster.ids)</span>
<span id="cb185-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-16" tabindex="-1"></a>Trem2_nuc_mac <span class="ot">&lt;-</span> <span class="fu">StashIdent</span>(<span class="at">object =</span> Trem2_nuc_mac, <span class="at">save.name =</span> <span class="st">&quot;MPC_Clusters&quot;</span>)</span>
<span id="cb185-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-17" tabindex="-1"></a></span>
<span id="cb185-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-18" tabindex="-1"></a><span class="co"># Reorder clusters</span></span>
<span id="cb185-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-19" tabindex="-1"></a><span class="fu">levels</span>(Trem2_nuc_mac)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;Resident_Mac&quot;</span>,</span>
<span id="cb185-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-20" tabindex="-1"></a>                        <span class="st">&quot;Mac_Cd209+&quot;</span>,</span>
<span id="cb185-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-21" tabindex="-1"></a>                        <span class="st">&quot;Foamy Mac&quot;</span>,</span>
<span id="cb185-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-22" tabindex="-1"></a>                        <span class="st">&quot;Mac_Tgfbr1&quot;</span>,</span>
<span id="cb185-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-23" tabindex="-1"></a>                        <span class="st">&quot;Mono&quot;</span>,</span>
<span id="cb185-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-24" tabindex="-1"></a>                        <span class="st">&quot;Cavity_Mac&quot;</span>,</span>
<span id="cb185-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-25" tabindex="-1"></a>                        <span class="st">&quot;cDC1&quot;</span>,</span>
<span id="cb185-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-26" tabindex="-1"></a>                        <span class="st">&quot;cDC2&quot;</span>,</span>
<span id="cb185-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-27" tabindex="-1"></a>                        <span class="st">&quot;Mature_DC&quot;</span>,</span>
<span id="cb185-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-28" tabindex="-1"></a>                        <span class="st">&quot;Myelo_Padi4+&quot;</span>,</span>
<span id="cb185-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-29" tabindex="-1"></a>                        <span class="st">&quot;nd&quot;</span>,</span>
<span id="cb185-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-30" tabindex="-1"></a>                        <span class="st">&quot;low_quality&quot;</span>)</span>
<span id="cb185-31"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-31" tabindex="-1"></a></span>
<span id="cb185-32"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-32" tabindex="-1"></a>new.cluster.ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;a_Resident_Mac&quot;</span>,</span>
<span id="cb185-33"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-33" tabindex="-1"></a>                     <span class="st">&quot;b_Mac_Cd209+&quot;</span>,</span>
<span id="cb185-34"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-34" tabindex="-1"></a>                     <span class="st">&quot;c_Foamy Mac&quot;</span>,</span>
<span id="cb185-35"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-35" tabindex="-1"></a>                     <span class="st">&quot;d_Mac_Tgfbr1&quot;</span>,</span>
<span id="cb185-36"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-36" tabindex="-1"></a>                     <span class="st">&quot;e_Mono&quot;</span>,</span>
<span id="cb185-37"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-37" tabindex="-1"></a>                     <span class="st">&quot;f_Cavity_Mac&quot;</span>,</span>
<span id="cb185-38"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-38" tabindex="-1"></a>                     <span class="st">&quot;g_cDC1&quot;</span>,</span>
<span id="cb185-39"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-39" tabindex="-1"></a>                     <span class="st">&quot;h_cDC2&quot;</span>,</span>
<span id="cb185-40"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-40" tabindex="-1"></a>                     <span class="st">&quot;i_Mature_DC&quot;</span>,</span>
<span id="cb185-41"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-41" tabindex="-1"></a>                     <span class="st">&quot;j_Myelo_Padi4+&quot;</span>,</span>
<span id="cb185-42"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-42" tabindex="-1"></a>                     <span class="st">&quot;k_nd&quot;</span>,</span>
<span id="cb185-43"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-43" tabindex="-1"></a>                     <span class="st">&quot;l_low_quality&quot;</span>)</span>
<span id="cb185-44"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-44" tabindex="-1"></a><span class="fu">names</span>(new.cluster.ids) <span class="ot">&lt;-</span> <span class="fu">levels</span>(Trem2_nuc_mac)</span>
<span id="cb185-45"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-45" tabindex="-1"></a>Trem2_nuc_mac<span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(Trem2_nuc_mac, new.cluster.ids)</span>
<span id="cb185-46"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-46" tabindex="-1"></a>Trem2_nuc_mac <span class="ot">&lt;-</span> <span class="fu">StashIdent</span>(<span class="at">object =</span> Trem2_nuc_mac, <span class="at">save.name =</span> <span class="st">&quot;MPC_Clusters&quot;</span>)</span>
<span id="cb185-47"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-47" tabindex="-1"></a></span>
<span id="cb185-48"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-48" tabindex="-1"></a><span class="co">#Extended Data Fig.6d</span></span>
<span id="cb185-49"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-49" tabindex="-1"></a><span class="co">#UMAPPlot(Trem2_nuc_mac)+scale_color_manual(values=c(&quot;#66C2A5&quot;, &quot;#C5A07A&quot; ,&quot;#DD927E&quot; ,&quot;#979EC1&quot;, </span></span>
<span id="cb185-50"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-50" tabindex="-1"></a><span class="co">#                                                    &quot;#BE93C6&quot;, &quot;#DB98AE&quot;, &quot;#B1C968&quot;, &quot;#CED843&quot;,</span></span>
<span id="cb185-51"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb185-51" tabindex="-1"></a><span class="co">#                                                    &quot;#FCD738&quot; ,&quot;#ECC978&quot;, &quot;#D2BD9F&quot;, &quot;#B3B3B3&quot;))</span></span></code></pre></div>
<div class="figure">
<img src="images/fig_6d.png" alt="Figure 8: UMAP plot of mononuclear phagocytes after subsetting and reclustering (Mac: macrophage; Myelo: myeloid cell; DC: dendritic cell; nd: not determined)" width="1337" />
<p class="caption">
(#fig:vergelijking figuur 6d)Figure 8: UMAP plot of mononuclear phagocytes after subsetting and reclustering (Mac: macrophage; Myelo: myeloid cell; DC: dendritic cell; nd: not determined)
</p>
</div>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb186-1" tabindex="-1"></a><span class="co">#Extended Data Fig.6e</span></span>
<span id="cb186-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb186-2" tabindex="-1"></a><span class="fu">DotPlot</span>(Trem2_nuc_mac, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;Mrc1&quot;</span>, <span class="st">&quot;F13a1&quot;</span>,<span class="st">&quot;Cd163&quot;</span>, <span class="st">&quot;Cd209f&quot;</span>, <span class="st">&quot;Cd209g&quot;</span>, <span class="st">&quot;Gpnmb&quot;</span>,</span>
<span id="cb186-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb186-3" tabindex="-1"></a>                                    <span class="st">&quot;Atp6v0d2&quot;</span>, <span class="st">&quot;Cd5l&quot;</span>,<span class="st">&quot;Ctsl&quot;</span>, <span class="st">&quot;Tgfbr1&quot;</span>, </span>
<span id="cb186-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb186-4" tabindex="-1"></a>                                    <span class="st">&quot;Mgat5&quot;</span>, <span class="st">&quot;Apobec1&quot;</span>, <span class="st">&quot;Plac8&quot;</span>, </span>
<span id="cb186-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb186-5" tabindex="-1"></a>                                    <span class="st">&quot;Ace&quot;</span>, <span class="st">&quot;Itgal&quot;</span>, <span class="st">&quot;Cd226&quot;</span>, <span class="st">&quot;Retnla&quot;</span>, <span class="st">&quot;Flt3&quot;</span>,<span class="st">&quot;Dpp4&quot;</span> ,</span>
<span id="cb186-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb186-6" tabindex="-1"></a>                                    <span class="st">&quot;Clec9a&quot;</span>, <span class="st">&quot;Cd209a&quot;</span>, <span class="st">&quot;Ccr7&quot;</span>, <span class="st">&quot;Padi4&quot;</span>),<span class="at">cols=</span><span class="st">&quot;RdYlBu&quot;</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<div class="figure">
<img src="images/fig_6e.png" alt="Figure 9: marker genes of mononuclear phagocyte clusters" width="1309" />
<p class="caption">
(#fig:vergelijking figuur 6e)Figure 9: marker genes of mononuclear phagocyte clusters
</p>
</div>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-1" tabindex="-1"></a><span class="do">## DE Markers in Foamy macs WT vs Trem2 directly calculated on single-cell data </span></span>
<span id="cb187-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-2" tabindex="-1"></a>Foamy<span class="ot">=</span><span class="fu">subset</span>(Trem2_nuc_mac, <span class="at">idents=</span><span class="st">&quot;c_Foamy Mac&quot;</span>)</span>
<span id="cb187-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-3" tabindex="-1"></a><span class="fu">Idents</span>(Foamy)<span class="ot">=</span><span class="st">&quot;Condition&quot;</span></span>
<span id="cb187-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-4" tabindex="-1"></a>Foamy_Markers<span class="ot">=</span><span class="fu">FindMarkers</span>(Foamy, <span class="at">ident.1 =</span> <span class="st">&quot;WT&quot;</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.2</span>)</span>
<span id="cb187-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-5" tabindex="-1"></a></span>
<span id="cb187-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-6" tabindex="-1"></a><span class="co"># DotPlot of selected DE markers</span></span>
<span id="cb187-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb187-7" tabindex="-1"></a><span class="fu">DotPlot</span>(Foamy, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;Msr1&quot;</span>, <span class="st">&quot;Cd36&quot;</span>,<span class="st">&quot;Spp1&quot;</span>, <span class="st">&quot;Cd5l&quot;</span>, <span class="st">&quot;Atp6v0a1&quot;</span>, <span class="st">&quot;Cd38&quot;</span>,<span class="st">&quot;Bcl2&quot;</span>),<span class="at">cols=</span><span class="st">&quot;RdYlBu&quot;</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/DE%20markers-1.png" width="672" /></p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-1" tabindex="-1"></a><span class="do">############### Perform Pseudobulk DE analysis in WT vs KO foamy macrophages ##################</span></span>
<span id="cb188-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-2" tabindex="-1"></a><span class="co"># Here we only included samples where &gt;200 foamy macrophages were recovered</span></span>
<span id="cb188-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-3" tabindex="-1"></a><span class="co"># to avoid potential impact of outlier cells on aggregated pseudobulk count matrices</span></span>
<span id="cb188-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-4" tabindex="-1"></a></span>
<span id="cb188-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-5" tabindex="-1"></a><span class="do">##*Create a separate object for each sample</span></span>
<span id="cb188-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-6" tabindex="-1"></a><span class="fu">Idents</span>(Foamy)<span class="ot">=</span><span class="st">&quot;Sample&quot;</span></span>
<span id="cb188-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-7" tabindex="-1"></a>Foamy_WT1<span class="ot">=</span><span class="fu">subset</span>(Foamy, <span class="at">idents =</span> <span class="st">&quot;WT_1&quot;</span>)</span>
<span id="cb188-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-8" tabindex="-1"></a>Foamy_WT2<span class="ot">=</span><span class="fu">subset</span>(Foamy, <span class="at">idents =</span> <span class="st">&quot;WT_2&quot;</span>)</span>
<span id="cb188-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-9" tabindex="-1"></a>Foamy_KO2<span class="ot">=</span><span class="fu">subset</span>(Foamy, <span class="at">idents =</span> <span class="st">&quot;KO_2&quot;</span>)</span>
<span id="cb188-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-10" tabindex="-1"></a>Foamy_KO3<span class="ot">=</span><span class="fu">subset</span>(Foamy, <span class="at">idents =</span> <span class="st">&quot;KO_3&quot;</span>)</span>
<span id="cb188-11"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-11" tabindex="-1"></a></span>
<span id="cb188-12"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-12" tabindex="-1"></a><span class="do">## Generate Pseudobulk count matrices</span></span>
<span id="cb188-13"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-13" tabindex="-1"></a>Foamy_WT1_PSB<span class="ot">=</span><span class="fu">GetAssayData</span>(<span class="at">object =</span> Foamy_WT1, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb188-14"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-14" tabindex="-1"></a>Foamy_WT1_PSB<span class="ot">=</span><span class="fu">as.data.frame</span>(Foamy_WT1_PSB)</span>
<span id="cb188-15"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-15" tabindex="-1"></a>Foamy_WT1_PSB<span class="ot">=</span><span class="fu">rowSums</span>(Foamy_WT1_PSB)</span>
<span id="cb188-16"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-16" tabindex="-1"></a>Foamy_WT1_PSB<span class="ot">=</span><span class="fu">as.matrix</span>(Foamy_WT1_PSB)</span>
<span id="cb188-17"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-17" tabindex="-1"></a><span class="fu">colnames</span>(Foamy_WT1_PSB)<span class="ot">=</span><span class="st">&quot;Foamy_WT1_PSB&quot;</span></span>
<span id="cb188-18"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-18" tabindex="-1"></a></span>
<span id="cb188-19"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-19" tabindex="-1"></a>Foamy_WT2_PSB<span class="ot">=</span><span class="fu">GetAssayData</span>(<span class="at">object =</span> Foamy_WT2, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb188-20"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-20" tabindex="-1"></a>Foamy_WT2_PSB<span class="ot">=</span><span class="fu">as.data.frame</span>(Foamy_WT2_PSB)</span>
<span id="cb188-21"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-21" tabindex="-1"></a>Foamy_WT2_PSB<span class="ot">=</span><span class="fu">rowSums</span>(Foamy_WT2_PSB)</span>
<span id="cb188-22"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-22" tabindex="-1"></a>Foamy_WT2_PSB<span class="ot">=</span><span class="fu">as.matrix</span>(Foamy_WT2_PSB)</span>
<span id="cb188-23"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-23" tabindex="-1"></a><span class="fu">colnames</span>(Foamy_WT2_PSB)<span class="ot">=</span><span class="st">&quot;Foamy_WT2_PSB&quot;</span></span>
<span id="cb188-24"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-24" tabindex="-1"></a></span>
<span id="cb188-25"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-25" tabindex="-1"></a>Foamy_KO2_PSB<span class="ot">=</span><span class="fu">GetAssayData</span>(<span class="at">object =</span> Foamy_KO2, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb188-26"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-26" tabindex="-1"></a>Foamy_KO2_PSB<span class="ot">=</span><span class="fu">as.data.frame</span>(Foamy_KO2_PSB)</span>
<span id="cb188-27"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-27" tabindex="-1"></a>Foamy_KO2_PSB<span class="ot">=</span><span class="fu">rowSums</span>(Foamy_KO2_PSB)</span>
<span id="cb188-28"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-28" tabindex="-1"></a>Foamy_KO2_PSB<span class="ot">=</span><span class="fu">as.matrix</span>(Foamy_KO2_PSB)</span>
<span id="cb188-29"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-29" tabindex="-1"></a><span class="fu">colnames</span>(Foamy_KO2_PSB)<span class="ot">=</span><span class="st">&quot;Foamy_KO2_PSB&quot;</span></span>
<span id="cb188-30"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-30" tabindex="-1"></a></span>
<span id="cb188-31"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-31" tabindex="-1"></a>Foamy_KO3_PSB<span class="ot">=</span><span class="fu">GetAssayData</span>(<span class="at">object =</span> Foamy_KO3, <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb188-32"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-32" tabindex="-1"></a>Foamy_KO3_PSB<span class="ot">=</span><span class="fu">as.data.frame</span>(Foamy_KO3_PSB)</span>
<span id="cb188-33"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-33" tabindex="-1"></a>Foamy_KO3_PSB<span class="ot">=</span><span class="fu">rowSums</span>(Foamy_KO3_PSB)</span>
<span id="cb188-34"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-34" tabindex="-1"></a>Foamy_KO3_PSB<span class="ot">=</span><span class="fu">as.matrix</span>(Foamy_KO3_PSB)</span>
<span id="cb188-35"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-35" tabindex="-1"></a><span class="fu">colnames</span>(Foamy_KO3_PSB)<span class="ot">=</span><span class="st">&quot;Foamy_KO3_PSB&quot;</span></span>
<span id="cb188-36"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-36" tabindex="-1"></a></span>
<span id="cb188-37"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-37" tabindex="-1"></a><span class="co"># Merge in a single count matrix**</span></span>
<span id="cb188-38"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-38" tabindex="-1"></a>counts_data<span class="ot">=</span><span class="fu">data.frame</span>(Foamy_WT1_PSB, Foamy_WT2_PSB, Foamy_KO2_PSB, Foamy_KO3_PSB)</span>
<span id="cb188-39"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-39" tabindex="-1"></a></span>
<span id="cb188-40"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-40" tabindex="-1"></a><span class="co"># Create the col_data dataframe for DESEQ2</span></span>
<span id="cb188-41"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-41" tabindex="-1"></a>Sample<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;WT_1&quot;</span>, <span class="st">&quot;WT_2&quot;</span>, <span class="st">&quot;KO_2&quot;</span>, <span class="st">&quot;KO_3&quot;</span>)</span>
<span id="cb188-42"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-42" tabindex="-1"></a>Group<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;WT&quot;</span>,<span class="st">&quot;KO&quot;</span>,<span class="st">&quot;KO&quot;</span>)</span>
<span id="cb188-43"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-43" tabindex="-1"></a>rownames<span class="ot">=</span><span class="fu">colnames</span>(counts_data)</span>
<span id="cb188-44"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-44" tabindex="-1"></a>col_data<span class="ot">=</span><span class="fu">data.frame</span>(Sample, Group, <span class="at">row.names =</span> rownames)</span>
<span id="cb188-45"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-45" tabindex="-1"></a></span>
<span id="cb188-46"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-46" tabindex="-1"></a><span class="do">## Check that all sample are there in both matrices</span></span>
<span id="cb188-47"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb188-47" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(counts_data) <span class="sc">%in%</span> <span class="fu">rownames</span>(col_data))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb190-1" tabindex="-1"></a><span class="do">## Check that the are in the same order</span></span>
<span id="cb190-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb190-2" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(counts_data) <span class="sc">==</span> <span class="fu">rownames</span>(col_data))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-1" tabindex="-1"></a><span class="do">## *Create DESEq2 object*</span></span>
<span id="cb192-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb192-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-3" tabindex="-1"></a>Foamy_PSB_DESeq2<span class="ot">=</span><span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts_data, <span class="at">colData =</span> col_data, <span class="at">design =</span> <span class="sc">~</span>Group)</span>
<span id="cb192-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-4" tabindex="-1"></a></span>
<span id="cb192-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-5" tabindex="-1"></a><span class="co"># Pre-filtering</span></span>
<span id="cb192-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-6" tabindex="-1"></a>keep<span class="ot">=</span><span class="fu">rowSums</span>(<span class="fu">counts</span>(Foamy_PSB_DESeq2))<span class="sc">&gt;=</span><span class="dv">3</span></span>
<span id="cb192-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-7" tabindex="-1"></a>Foamy_PSB_DESeq2<span class="ot">=</span>Foamy_PSB_DESeq2[keep,]</span>
<span id="cb192-8"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-8" tabindex="-1"></a></span>
<span id="cb192-9"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-9" tabindex="-1"></a><span class="co"># Set the factor level</span></span>
<span id="cb192-10"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb192-10" tabindex="-1"></a><span class="fu">relevel</span>(Foamy_PSB_DESeq2<span class="sc">$</span>Group, <span class="at">ref =</span> <span class="st">&#39;WT&#39;</span>)</span></code></pre></div>
<pre><code>## [1] WT WT KO KO
## Levels: WT KO</code></pre>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb194-1" tabindex="-1"></a><span class="co"># Run DESeq</span></span>
<span id="cb194-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb194-2" tabindex="-1"></a>Foamy_PSB_DESeq2<span class="ot">=</span><span class="fu">DESeq</span>(Foamy_PSB_DESeq2)</span>
<span id="cb194-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb194-3" tabindex="-1"></a>res<span class="ot">=</span><span class="fu">results</span>(Foamy_PSB_DESeq2, <span class="at">alpha =</span> <span class="fl">0.1</span>)</span>
<span id="cb194-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb194-4" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>## 
## out of 17503 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 61, 0.35%
## LFC &lt; 0 (down)     : 22, 0.13%
## outliers [1]       : 0, 0%
## low counts [2]     : 12895, 74%
## (mean count &lt; 120)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb196-1" tabindex="-1"></a><span class="fu">plotMA</span>(res)</span></code></pre></div>
<p><img src="_main_files/figure-html/DE%20markers%20pt%202-1.png" width="672" /></p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-1" tabindex="-1"></a><span class="co"># Plot selected DE genes with pheatmap (Extended Data Fig. 6f)</span></span>
<span id="cb197-2"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-2" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>( Foamy_PSB_DESeq2, <span class="at">fitType=</span><span class="st">&#39;mean&#39;</span>, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb197-3"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-3" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb197-4"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-4" tabindex="-1"></a>selected_genes<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;Hmox1&quot;</span>, <span class="st">&quot;Cd38&quot;</span>, <span class="st">&quot;Msr1&quot;</span>, <span class="st">&quot;Bcl2&quot;</span>)</span>
<span id="cb197-5"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-5" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">assay</span>(rld)[selected_genes,], <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>)</span>
<span id="cb197-6"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-6" tabindex="-1"></a></span>
<span id="cb197-7"><a href="free-space-visualisation-of-scrna-sequencing-using-python.html#cb197-7" tabindex="-1"></a><span class="do">#######end</span></span></code></pre></div>
<div class="figure">
<img src="images/fig_6f.png" alt="Figure 10: expression of selected differentially expressed genes in foamy macrophages as determined from pseudo-bulk analysis of experimental replicates from the snRNA-seq data. Samples with &gt;200 foamy macrophages were included for pseudo-bulk analysis." width="1144" />
<p class="caption">
(#fig:vergelijking figuur 6f)Figure 10: expression of selected differentially expressed genes in foamy macrophages as determined from pseudo-bulk analysis of experimental replicates from the snRNA-seq data. Samples with &gt;200 foamy macrophages were included for pseudo-bulk analysis.
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cv.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="guerrilla-analytics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
